clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -66,52 +66,51 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals) {
         this(definition, originals, true);
     }
 
-    protected Object get(String key) {
+    protected <T> T get(String key) {
```

Comment:
We were doing explicit casts in those cases, right? I agree that it would be better to check before casting if that's what you are suggesting.